File: src/main/java/com/github/thorben/webnd/domain/user/Username.java
package com.github.thorben.webnd.domain.user;

import com.github.thorben.webnd.foundation.AbstractValueObject;

import javax.persistence.Embeddable;

@Embeddable
public class Username extends AbstractValueObject<String> {

	protected Username() {
		// JPA
	}

	public Username(String value) {
		super(value);
	}
}

File: src/main/kotlin/com/github/thorben/webnd/domain/user/Username.kt
package com.github.thorben.webnd.domain.user

import com.github.thorben.webnd.foundation.AbstractValueObject
import javax.persistence.Embeddable

@Embeddable
class Username(value: String) : AbstractValueObject<String>(value) {
	protected constructor(): this("") {
		// JPA
	}
}